packages/lib-utils/src/k8s/hooks/watch-resource-types.ts

```diff
@@ -0,0 +1,59 @@
+import type {
+  K8sGroupVersionKind,
+  K8sResourceCommon,
+  K8sResourceKindReference,
+  Selector,
+} from '../../types/k8s';
+
+export type WatchK8sResult<R extends K8sResourceCommon | K8sResourceCommon[]> = [
```

Comment:
Still need to clean up the type situation present in the SDK today (in root type folder versus next to the consuming JS). I'm fine with this for now!

Reply:
Yeah, I did modify any existing structure, just co-located stuff as I saw fit for the objective -- based on that conversation